packages/core/src/common/strings.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+export function* split(s: string, splitter: string): IterableIterator<string> {
```

Comment:
What's the use case for this here? Wouldn't the usual string.split('\n') work in this case? Or is it better because it just returns the first splitted string and doesn't do it all in one go?

Reply:
it let's iterate over lines without holding an array of all lines in memory. you can scan file content more efficient, especially in cases where you might interrupt the iteration in between.

Reply:
Ah that makes sense, thanks for the explanation!